### PR TITLE
KaizenTask1052_Unit_test_populate_exchange_currency_tables

### PR DESCRIPTION
**Pull request by heanhsok**
Task #1052

- Mock CCXT.
- Verify that the exchange_currency and currency_pair tables are populated correctly.

### CODE REVIEW

**Comment by heanhsok** [comment]
I didn't include the mock db connection part because it looks like the `populate_exchange_currency_tables` function doesn't make use of `DbConnectionManager` from `im_v2.common.db.db_utils`. Please let me know if I miss something. @sonaalKant

File: im_v2/ccxt/db/test/test_ccxt_db_utils.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    reason="Need docker children / sibling support and/or Run only if CK S3 is available",
+)
+@pytest.mark.slow("22 seconds.")
+class TestPopulateExchangeCurrencyTables(imvcddbut.TestImDbHelper):
+    @classmethod
+    def get_id(cls) -> int:
+        return hash(cls.__name__) % 10000
+
+    # Mock calls to external providers.
+    ccxt_patch = umock.patch.object(imvccdbut, "ccxt", spec=ccxt)
```
Comment (DanilYachmenev) [inline im_v2/ccxt/db/test/test_ccxt_db_utils.py:28]: put it before all the functions, for ref see `im_v2/ccxt/data/extract/test/test_download_exchange_data_to_db_periodically.py`

File: im_v2/ccxt/db/test/test_ccxt_db_utils.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    # This will be run before and after each test.
+    @pytest.fixture(autouse=True)
+    def setup_teardown_test(self):
+        # Run before each test.
+        self.set_up_test()
+        yield
+        # Run after each test.
+        self.tear_down_test()
```
Comment (DanilYachmenev) [inline im_v2/ccxt/db/test/test_ccxt_db_utils.py:37]: not really our coding style but following reference at `amp/im_v2/ccxt/data/extract/test/test_download_exchange_data_to_db_periodically.py`

no need to do anything here, just a note for other reviewers

File: im_v2/ccxt/db/test/test_ccxt_db_utils.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        self.ccxt_mock: umock.MagicMock = self.ccxt_patch.start()
+        # Set up the mock exchanges
+        self.ccxt_mock.exchanges = ["binance"]
+        # Mock the binance ccxt to return load market keys.
+        self.ccxt_mock.binance.return_value.load_markets.return_value.keys.return_value = [
+            "MOCKBTC/MOCKUSD"
+        ]
+
+    def tear_down_test(self) -> None:
+        # Drop `exchange_name` table in the database.
```
Comment (DanilYachmenev) [inline im_v2/ccxt/db/test/test_ccxt_db_utils.py:60]: no need for this comment imo, line is self-explanatory

File: im_v2/ccxt/db/test/test_ccxt_db_utils.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        hsql.remove_table(self.connection, "exchange_name")
+        # Drop `currency_pair` table in the database.
```
Comment (DanilYachmenev) [inline im_v2/ccxt/db/test/test_ccxt_db_utils.py:62]: same

File: im_v2/ccxt/db/test/test_ccxt_db_utils.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        self.ccxt_patch.stop()
+
+    def test1(self) -> None:
+        """
+        Verify that the exchange_currency and currency_pair tables are
+        populated correctly.
+        """
+        # Run test.
+        imvccdbut.populate_exchange_currency_tables(self.connection)
+        #
```
Comment (DanilYachmenev) [inline im_v2/ccxt/db/test/test_ccxt_db_utils.py:74]: what are these empty lines encircling a comment?
no need

File: im_v2/ccxt/db/test/test_ccxt_db_utils.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        # Verify that the exchange_currency table is populated correctly.
```
Comment (DanilYachmenev) [inline im_v2/ccxt/db/test/test_ccxt_db_utils.py:75]: if you refer an object name - put it in backticks
but in general avoid it at all cost
https://github.com/kaizen-ai/kaizenflow/blob/master/docs/coding/all.coding_style.how_to_guide.md#referring-to-an-object-in-code-comments

File: im_v2/ccxt/db/test/test_ccxt_db_utils.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+        self.check_df_output(
+            actual,
+            expected_length,
+            expected_column_value,
+            expected_column_value,
+            expected_signature,
+        )
+        #
+        # Verify that the currency_pair table is populated correctly.
+        #
```
Comment (DanilYachmenev) [inline im_v2/ccxt/db/test/test_ccxt_db_utils.py:101]: same

File: im_v2/ccxt/db/test/test_ccxt_db_utils.py [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    def test1(self) -> None:
+        """
+        Verify that the exchange names and currency pairs tables are populated
+        correctly.
+        """
+        # Run test.
+        imvccdbut.populate_exchange_currency_tables(self.connection)
+        # Verify that the exchange names table is populated correctly.
+        # Check the content of the table.
+        query = f"SELECT * FROM exchange_currency"
```
Comment (sonaalKant) [inline im_v2/ccxt/db/test/test_ccxt_db_utils.py:74]: it should be `exchange_name`

**Review by sonaalKant — CHANGES_REQUESTED** [review]
Awesome Job!!. Test case passed, just one small fix required.

**Review by sonaalKant — APPROVED** [review]

**Comment by heanhsok** [comment]
what's the next step? should we merge this PR?

**Review by gpsaggese — COMMENTED** [review]
Well done.

**Pull request merged**